### PR TITLE
DEV-2612 kinesis client changes

### PR DESCRIPTION
**Pull request by zsid60**
image

### CODE REVIEW

**Comment by zsid60** [comment]
@nikhilgarg28 I verified that the datetime -> proto.Timestamp conversion behaves as expected with timezones

**Review by nikhilgarg28 — APPROVED** [review]

File: fennel/lib/to_proto/to_proto.py [hunk 1]
```diff
@@ -1173,25 +1173,21 @@ def _kinesis_to_ext_db_proto(
 def _kinesis_to_ext_table_proto(
     db: connector_proto.ExtDatabase,
     stream_arn: str,
-    init_position: sources.InitPosition,
-    init_timestamp: Optional[datetime] = None,
+    init_position: str | kinesis.at_timestamp,
     format: str = "json",
```
Comment (mohitreddy1996) [inline fennel/lib/to_proto/to_proto.py:1177]: should we remove default value from here too and add a validation that format must be json?

**Review by mohitreddy1996 — APPROVED** [review]

File: fennel/lib/to_proto/to_proto.py (hunk 1 above)
Comment (mohitreddy1996) [reply fennel/lib/to_proto/to_proto.py:1177]: We only need to make the argument on the public API optional and have a default value. As the value propagates down the code to protos, we shouldn't have default values? But rather add validation that it is not json?

Comment (zsid60) [reply fennel/lib/to_proto/to_proto.py:1177]: the validation is already present and tested but good catch, will remove this here

Comment (mohitreddy1996) [reply fennel/lib/to_proto/to_proto.py:1177]: I don't think there is a validation added here in this method (to_proto.py file for this). Please add one and add an unit test for it in test_to_proto.py (this future proofs that if there is a different caller for this, which there could easily be, fails if they pass an invalid format).

**Pull request merged**